### ISSUE TITLE
Panels Slow 

### ISSUE THREAD

**Issue by rudgej** [Blackymas/NSPanel_HA_Blueprint#1203]
I have 15 panels in my system  of which currently 10 are active ( the rest are powered off).

The system is running on a PI4.

The panels are very slow to display the devices when I swipe left. 

I use the front buttons to trigger the most common scenes and these are now becoming unreliable. 

I can see that the automation runs every minute on all panels - and they all seem to run at the same time as opposed to being staggered over that minute. 

I am not sure why the automation runs as I assumed the buttons etc are held on the panel and then trigger a service when pressed?

I don't know how many others are running large panel numbers ( I have one in every room as a scene controller) and if there are any performance enhancement I can try.

I will move to an Intel NUC shortly as the size of the system is quite large ( 70 dimmers) and the processor seems ti ideal about about 20% and peak at 50%. There is also an obvious memory leak as memory keeps on growing but I doubt this is anything to do with this automation,.

**Comment by edwardtfn** [comment]
Wow! 15 panels!!
We definitely can find some improvement in the code for reducing the load in your Home Assistant server and in the network. The latest version already moved quite a lot of processing to ESPHome, but still some basic things in the Blueprint side (like the date string creation, which is dine every 1min and certainly can be changed to once a day).
I can take a look at this later.
The Home page have most of its components updated even when that page is not visible. This helps with performance for smaller system, but it could be an issue for large systems like yours. Anyways, if you are now using chips and the values on the home page, that will be updated every time that value is updated, regardless if that page is shown. The other pages are updated only when visible.

By the way, on the boot page, which versions you have for TFT, ESPHome and Blueprint?

**Comment by edwardtfn** [comment]
Ah, and I've moved my HA from a RPi4 to a x86 with a SSD disk around an year ago and the performance improvement is definitely visible, although I didn't had major issues with the RPi4 other than when doing heavy tasks, like compiling ESPHome.

**Comment by rudgej** [comment]
Hi thank you for the reply. 

There are no chips etc on the home page just the button screens. The chips I have not really got to grips with - they seem to just show status. 

To be honest 99% of the time we use the front hardware buttons to set a scene. Left off. Long push hi. Right " normal " long push medium. 

Even then automation does most of the work so the panels are just a nicer labeled version of the light switch.

**Comment by edwardtfn** [comment]
I found an issue where if an entity used in any of the button pages are updated, the automation would run without sending anything to the panel, but adding an entry to the Logbook, which could spam the system with unnecessary (or unwanted) messages and for large systems could drive to some performance degradation.
Also, I've added a delay to the panel update when automations are reloaded. Until v4.0.2, all panels will receive all the settings at the same time when the automations are reloaded (or HA starts). Now it will reserve a slot of a few seconds per panel, distributing the calls and reducing the chances for overloading the system.
All those changes are on `dev` now and will be made available with the next release.
